python/erdos/__init__.py

```diff
@@ -83,6 +84,17 @@ def connect(
     node_id = _num_py_operators
     logger.debug("Connecting operator #{num} ({name}) to the graph.".format(
         num=node_id, name=config.name))
+    if config._address is not None:
+        global node_to_address
+        addresses = list(node_to_address.values())
+        processes_per_address = {a: addresses.count(a) for a in set(addresses)}
+        if len(node_to_address) == 0:
+            processes_per_address[config._address] = 1
+            node_to_address[0] = (config._address, 1)
+        processes_per_address[config._address] = \
+            processes_per_address.get(config._address, 0) + 1
+        node_to_address[node_id] = (config._address,
+                                    processes_per_address[config._address])
```

Comment:
I'm not sure about this entire section -- I think it could be done in a clearer way. See my review comment.